docs/runs.md

```diff
@@ -139,20 +139,59 @@ the `example.dev` API group,  with the version `v1alpha1`.
 #### Developer guide for custom controllers supporting `spec`.
 
 1. A custom controller may or may not support a `Spec`. In cases where it is
-   not supported the custom controller should respond with proper validation error.
-
-2. Validation of the fields of the custom task is delegated to the custom task controller. It is recommended to
-   implement validations as asynchronous
-   (i.e. at reconcile time), rather than part of the webhook. Using a webhook for validation is problematic because, it
-   is not possible to filter custom task resource objects before validation step, as a result each custom task resource
-   has to undergo validation by all the installed custom task controllers.
-   
+   not supported the custom controller should respond with proper validation
+   error.
+
+2. Validation of the fields of the custom task is delegated to the custom
+   task controller. It is recommended to implement validations as asynchronous
+   (i.e. at reconcile time), rather than part of the webhook. Using a webhook
+   for validation is problematic because, it is not possible to filter custom
+   task resource objects before validation step, as a result each custom task
+   resource has to undergo validation by all the installed custom task
+   controllers.
+
 3. A custom task may have an empty spec, but cannot have an empty
-    `ApiVersion` and `Kind`. Custom task controllers should handle
+   `ApiVersion` and `Kind`. Custom task controllers should handle
    an empty spec, either with a default behaviour, in a case no default
    behaviour is supported then, appropriate validation error should be
    updated to the `Run`'s status.
 
+### Specifying `Timeout`
+
+A custom task specification can be created with `Timeout` as follows:
+
+```yaml
+apiVersion: tekton.dev/v1alpha1
+kind: Run
+metadata:
+  generateName: simpleexample
+spec:
+  timeout: 10s # set timeouts here.
+  params:
+    - name: searching
+      value: the purpose of my existence
+  ref:
+    apiVersion: custom.tekton.dev/v1alpha1
+    kind: Example
+    name: exampleName
+```
+
+Supporting timeouts is optional but recommended.
+
+#### Developer guide for custom controllers supporting `Timeout`
+
+1. Tekton controllers will never directly update the status of the
+   `Run`, it is the responsibility of the custom task controller to support
```

Comment:
Am I right that what we mean here is that the custom task developer is expected to update the status/condition of the `Run`? It might be worth describing a simple common status/condition/reason that is used for timeouts. I'm thinking something like:

> 4. Once resources or timers are cleaned up it is good practice to set a
> condition on the `Run`'s `status` of `Succeeded/False` with a `Reason`
> of `RunTimedOut`.